src/deadline/unreal_submitter/unreal_open_job/unreal_open_job_step.py

```diff
@@ -0,0 +1,468 @@
+
+        if self._render_args_type == RenderUnrealOpenJobStep.RenderArgsType.NOT_SET:
+            raise exceptions.RenderArgumentsTypeNotSetError(
+                "RenderOpenJobStep parameters are not valid. Expect one of the following:\n"
+                f"- {OpenJobStepParameterNames.QUEUE_MANIFEST_PATH}\n"
+                f"- {OpenJobStepParameterNames.MOVIE_PIPELINE_QUEUE_PATH}\n"
+                f"- ({OpenJobStepParameterNames.LEVEL_SEQUENCE_PATH}, "
+                f"{OpenJobStepParameterNames.LEVEL_PATH}, "
+                f"{OpenJobStepParameterNames.MRQ_JOB_CONFIGURATION_PATH})\n"
+            )
+
+        task_chunk_id_param_definition = UnrealOpenJobStepParameterDefinition(
+            OpenJobStepParameterNames.TASK_CHUNK_ID,
+            TaskParameterType.INT.value,
+            [i for i in range(self._get_chunk_ids_count())],
+        )
+        self._update_extra_parameter(task_chunk_id_param_definition)
+
+        handler_param_definition = UnrealOpenJobStepParameterDefinition(
+            OpenJobStepParameterNames.ADAPTOR_HANDLER, TaskParameterType.STRING.value, ["render"]
+        )
+        self._update_extra_parameter(handler_param_definition)
+
+        if self.mrq_job:
+            output_setting = self.mrq_job.get_configuration().find_setting_by_class(
+                unreal.MoviePipelineOutputSetting
+            )
+            output_path = output_setting.output_directory.path
+            path_context = common.get_path_context_from_mrq_job(self.mrq_job)
+            output_path = output_path.format_map(path_context).rstrip("/")
+            output_param_definition = UnrealOpenJobStepParameterDefinition(
+                OpenJobStepParameterNames.OUTPUT_PATH, TaskParameterType.PATH.value, [output_path]
+            )
+            self._update_extra_parameter(output_param_definition)
+
+            if self._render_args_type == RenderUnrealOpenJobStep.RenderArgsType.QUEUE_MANIFEST_PATH:
+                manifest_param_definition = UnrealOpenJobStepParameterDefinition(
+                    OpenJobStepParameterNames.QUEUE_MANIFEST_PATH,
+                    TaskParameterType.PATH.value,
+                    [self._save_manifest_file()],
+                )
+                self._update_extra_parameter(manifest_param_definition)
+
+        step_entity = super()._build_template()
+
+        return step_entity
+
+    def _save_manifest_file(self):
+        new_queue = unreal.MoviePipelineQueue()
+        new_job = new_queue.duplicate_job(self.mrq_job)
+
+        # In duplicated job remove empty auto-detected files since
+        # we don't want them to be saved in manifest
+        new_job.preset_overrides.job_attachments.input_files.auto_detected = (
+            unreal.DeadlineCloudFileAttachmentsArray()
+        )
+
+        _, manifest_path = unreal.MoviePipelineEditorLibrary.save_queue_to_manifest_file(new_queue)
+        serialized_manifest = unreal.MoviePipelineEditorLibrary.convert_manifest_file_to_string(
```

Comment:
Unreal method unreal.Paths.create_temp_filename allows to create temp files with unique postfix. We need it for creating different manifests for different jobs since only one job added to each manifest